lib/mongo/auth/stringprep.rb

```diff
@@ -104,7 +96,12 @@ def mapping(c, mappings)
       end
 
       def normalize!(out)
-        out.unicode_normalize!(:nfkc)
+        if RUBY_VERSION < '2.2.0'
+          require 'mongo/auth/stringprep/unicode_normalize/normalize'
```

Comment:
You may consider only requiring this if `::UnicodeNormalize` is not an existing constant.

Reply:
Good idea. That covers alternative implementations that might add it in a different version. For example, does RUBY_VERSION < '2.2.0' also cover JRuby?